src/test/java/com/teragrep/cfe_39/Ingestion1Old1NewFileTest.java

```diff
@@ -0,0 +1,199 @@
+            MiniDFSCluster.Builder builder = new MiniDFSCluster.Builder(conf);
+            hdfsCluster = builder.build();
+            String hdfsURI = "hdfs://localhost:" + hdfsCluster.getNameNodePort() + "/";
+            config.setHdfsuri(hdfsURI);
+            DistributedFileSystem fileSystem = hdfsCluster.getFileSystem();
+
+            // ====== Init HDFS File System Object
+            Configuration fsConf = new Configuration();
+            // Set FileSystem URI
+            fsConf.set("fs.defaultFS", hdfsURI);
+            // Because of Maven
+            fsConf.set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem.class.getName());
+            fsConf.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
+            // Set HADOOP user
+            System.setProperty("HADOOP_USER_NAME", "hdfs");
+            System.setProperty("hadoop.home.dir", "/");
+            fs = FileSystem.get(URI.create(hdfsURI), fsConf);
+            // Inserts pre-made avro-files to HDFS where one file has new timestamp and other old, which are normally generated during data ingestion from mock kafka consumer.
+            String path = config.getHdfsPath() + "/" + "testConsumerTopic"; // "hdfs:///opt/teragrep/cfe_39/srv/testConsumerTopic"
+            // Sets the directory where the data should be stored, if the directory doesn't exist then it's created.
+            Path newDirectoryPath = new Path(path);
+            // Create new Directory
+            fs.mkdirs(newDirectoryPath);
+            LOGGER.debug("Path {} created.", path);
+            String dir = System.getProperty("user.dir") + "/src/test/java/com/teragrep/cfe_39/mockHdfsFiles";
+            Set<String> listOfFiles = Stream
+                    .of(Objects.requireNonNull(new File(dir).listFiles()))
+                    .filter(file -> !file.isDirectory())
+                    .map(File::getName)
+                    .collect(Collectors.toSet());
+            // Loop through all the avro files
+            for (String fileName : listOfFiles) {
+                String pathname = dir + "/" + fileName;
+                File avroFile = new File(pathname);
+                //==== Write file
+                LOGGER.debug("Begin Write file into hdfs");
+                //Create a path
+                Path hdfswritepath = new Path(newDirectoryPath + "/" + avroFile.getName()); // filename should be set according to the requirements: 0.12345 where 0 is Kafka partition and 12345 is Kafka offset.
+                Assertions.assertFalse(fs.exists(hdfswritepath));
+                Path readPath = new Path(avroFile.getPath());
+                fs.copyFromLocalFile(readPath, hdfswritepath);
+                LOGGER.debug("End Write file into hdfs");
+                LOGGER.debug("\nFile committed to HDFS, file writepath should be: {}\n", hdfswritepath);
+            }
+            fs.setTimes(new Path("hdfs:/opt/teragrep/cfe_39/srv/testConsumerTopic/0.9"), 157784760000L, -1);
```

Comment:
Any special meaning on these timestamps?

Reply:
This test case is for testing the functionality of the ingestion when there are correctly stored records/files already present in the database at the start of the ingestion.
Pruning of the HDFS database is implemented based on the file timestamp. In this specific test scenario one of the two files that already exist in HDFS database has old enough timestamp that it is going to be pruned from the database during the ingestion.